satpy/composites/__init__.py

```diff
@@ -1539,3 +1539,67 @@ def __call__(self, projectables, *args, **kwargs):
         res = super(BackgroundCompositor, self).__call__(data, **kwargs)
         res.attrs.update(attrs)
         return res
+
+
+class MaskingCompositor(GenericCompositor):
+    """A compositor that masks e.g. IR 10.8 channel data using cloud products from NWC SAF."""
+
+    def __init__(self, name, transparency=None, **kwargs):
+        """Collect custom configuration values.
+
+        Args:
+            transparency: transparency for each cloud type as key-value pairs
+                          in a dictionary
+
+        The transparencies can be either the numerical values in the
+        data used as a mask with the corresponding transparency
+        (0...100 %) as the value, or, for NWC SAF products, the flag
+        names in the dataset `flag_meanings` attribute.
+
+        For the values not listed in `transparencies`, the data will
+        be completely opaque (transparency = 0).
+
+        """
+        if transparency is None:
+            raise ValueError("No transparency configured for simple masking compositor")
+        self.transparency = transparency
+
+        super(MaskingCompositor, self).__init__(name, **kwargs)
+
+    def __call__(self, projectables, *args, **kwargs):
+        """Call the compositor."""
+        if len(projectables) != 2:
+            raise ValueError("Expected 2 datasets, got %d" % (len(projectables),))
+        projectables = self.match_data_arrays(projectables)
+        cloud_mask = projectables[1]
+        cloud_mask_data = cloud_mask.data
+        data = projectables[0]
+        alpha_attrs = data.attrs.copy()
+        if 'bands' in data.dims:
+            data = [data.sel(bands=b) for b in data['bands'] if b != 'A']
+        else:
+            data = [data]
+
+        # Create alpha band
+        alpha = da.ones((data[0].sizes['y'],
+                         data[0].sizes['x']),
+                        chunks=data[0].chunks)
+
+        # Modify alpha based on transparency per class from yaml
+        flag_meanings = cloud_mask.attrs['flag_meanings']
+        flag_values = cloud_mask.attrs['flag_values']
+
+        if isinstance(flag_meanings, str):
+            flag_meanings = flag_meanings.split()
+
+        for key, val in self.transparency.items():
+            if isinstance(key, str):
+                key_index = flag_meanings.index(key)
+                key = flag_values[key_index]
+            alpha_val = 1. - val / 100.
+            alpha = da.where(cloud_mask_data == key, alpha_val, alpha)
```

Comment:
Is this performing any computation ?

Reply:
would `choose` be an alternative ?

Reply:
No computations are triggered. To enforce this I added a custom scheduler (from Pyresample) and use it in the tests.

I'll have a look at `choose`.

Reply:
For me `choose` doesn't seem to be usable here. It would require a creation of a list of `N` arrays for the `choices`. The `N` would be the unique values present in the input mask, which isn't necessary to know. We typically want to only set few values with transparencies and leave the rest as opaque.